### ISSUE TITLE
Data samples keep moving with coordinate frame in 3D-Visualization

### ISSUE THREAD

**Issue by skasperski** [rock-core/gui-vizkit3d#29]
Data samples that are defined in some coordinate frame keep moving in the visualization when the frame changes over time. But data should stay at the position where it was recorded instead of moving with the frame after it has been recorded.

**Comment by jmachowinski** [comment]
Depends on the usecase, If you display a robot arm , you want that the 'subelements' move if you change the frame over time.

**Comment by skasperski** [comment]
Yes, that's true. But every reading of the outside world (e.g. pointclouds) you would expect to stay in place when the robot moves.

**Comment by skasperski** [comment]
I tried to attach the vis-plugin to a fixed frame node instead of the moving transformer tree node, but this doesn't seem to work. (Produces an indeed fixed, but strangely doubled and rotated visualization).

    void Vizkit3DWidget::enableDataHandler(VizPluginBase *viz)
    {
        if (viz == env_plugin)
            throw std::invalid_argument("attempted to enable the environment plugin");

        PluginMap::iterator it = plugins.find(viz);
        if (it != plugins.end())
        {            
            //get global position of framegroup
            osg::Group* framegroup = it->second.get();
            fixed_node = GlobalPos::getGlobalPATransform(framegroup, getRootNode());

            //need to save trans for disableDataHandler
            fixed_node->addChild(viz->getRootNode());
            getRootNode()->addChild(fixed_node);
        }
    }

**Comment by jmachowinski** [comment]
Did you try the keep_old_data option ?

**Comment by doudou** [comment]
@skasperski: I agree with you that it's a bug, but I don't think that anyone would have the time/energy to fix this particular issue unfortunately.

**Comment by pierrewillenbrockdfki** [comment]
@skasperski  do you think this is solved and can be closed?

**Comment by planthaber** [comment]
I guess it is meant that the e.g. point clouds displayed are not moving with their viz frame until the next cloud arrives

This cannot be solved vizkit3d as it doesn't really know when updateData is called on the plugin.

There are helper functions now in vitkit3d: https://github.com/rock-core/gui-vizkit3d/pull/73, but the actual solution must be implemented in the plugins updateData Function like e.g. here:

https://github.com/rock-gui/gui-vizkit3d_pcl/blob/master/viz/PCLPointCloudVisualization.hpp#L23

https://github.com/rock-gui/gui-vizkit3d_pcl/blob/master/viz/PCLPointCloudVisualization.hpp#L72

https://github.com/rock-gui/gui-vizkit3d_pcl/blob/master/viz/PCLPointCloudVisualization.cpp#L80-L83
